src/printer.js

```diff
@@ -1558,7 +1559,11 @@ function genericPrintNoParens(path, options, print) {
     case "DeclareTypeAlias":
     case "TypeAlias": {
       const parent = path.getParentNode(1);
-      if (parent && parent.type === "DeclareModule") {
+      if (
+        (parent && parent.type === "DeclareModule") ||
+          n.type === "DeclareTypeAlias" ||
+          (options.parser === "flow" && /declare/.test(getNodeSource(n, text)))
```

Comment:
Can you do `source.startsWith("declare")`, this way it's going to have less false positives if you happen to have named the type `declareSomething` or have written `declare` somewhere in the type declaration.